### PR TITLE
test: get kvstore and bugtool output

### PR DESCRIPTION
**Pull request by ianvernon**
* Get output of `cilium kvstore get cilium --recursive` in both runtime and
Kubernetes test suites.
* Get debugtool output and bugtool archive for each Cilium pod in the
Kubernetes test suite.

Signed-off by: Ian Vernon <ian@cilium.io>

Fixes: #2746 
Fixes: #3006

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

File: test/helpers/cons.go [hunk 1]
```diff
@@ -148,28 +148,31 @@ const (
...
-	"cilium endpoint list -o json":    "endpoint_list.txt",
-	"cilium service list -o json":     "service_list.txt",
-	"cilium config":                   "config.txt",
-	"sudo cilium bpf lb list":         "bpf_lb_list.txt",
-	"sudo cilium bpf ct list global":  "bpf_ct_list.txt",
-	"sudo cilium bpf tunnel list":     "bpf_tunnel_list.txt",
-	"cilium policy get":               "policy_get.txt",
-	"cilium status --all-controllers": "status.txt",
-	"sudo cilium debuginfo":           "debuginfo.txt",
+	"cilium endpoint list -o json":          "endpoint_list.txt",
```
Comment (eloycoto) [inline test/helpers/cons.go:151]: Hi, 

I do not liked the JSON output here. We should be able to debug endpoints, services as a user will do. The output of `-json` is a bit different, and we should be able to know problems without getting the JSON output. I'm happy if you add a new line without JSON output.

Comment (ianvernon) [reply test/helpers/cons.go:151]: I didn't change this, this was just due to a reformatting with `go fmt`. The output of `cilium endpoint list` is contained in the output of `debuginfo.txt`, which should be one of the primary sources for data so we can dogfood it.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

File: test/helpers/kubectl.go [hunk 2]
```diff
@@ -697,33 +692,22 @@ func (kub *Kubectl) CiliumReportDump(namespace string) {
...
-					logger.WithError(err).Errorf(
-						"cannot create test results for command '%s'", command)
+		// Get bugtool output. Since bugtool output is dumped in the pod's filesystem,
+		// copy it over with `kubectl cp`.
+		bugtoolCmd := fmt.Sprintf("kubectl exec -n %s %s -- cilium-bugtool", namespace, pod)
+		res := kub.Exec(bugtoolCmd)
+		if res.WasSuccessful() {
+			// Default output directory is /tmp for bugtool.
+			res = kub.Exec(fmt.Sprintf("kubectl exec -n %s %s -- ls /tmp/", namespace, pod))
+			tmpList := strings.Split(strings.TrimSuffix(res.stdout.String(), "\n"), "\n")
```
Comment (eloycoto) [inline test/helpers/kubectl.go]: Here you can use `res.ByLines` no?

File: test/helpers/cons.go (hunk 1 above)
Comment (eloycoto) [reply test/helpers/cons.go:151]: ok!

**Comment by ianvernon** [comment]
test-me-please

File: test/helpers/kubectl.go (hunk 2 above)
Comment (ianvernon) [reply test/helpers/kubectl.go]: Done.

File: test/helpers/cilium.go [hunk 3]
```diff
@@ -551,7 +551,7 @@ func (s *SSHMeta) DumpCiliumCommandOutput() {
 
 	// No need to create file for bugtool because it creates an archive of files
 	// for us.
-	res := s.ExecCilium(fmt.Sprintf("bugtool -t %s", filepath.Join(BasePath, testPath)))
```
Comment (tgraf) [inline test/helpers/cilium.go:554]: Sorry, I didn't see this. I grep'ed for `cilium bugtool` and didn't find anything.

**Review by tgraf — APPROVED** [review]

**Pull request merged**